### PR TITLE
[FIX] base, test_read_group : process duplicated groupbys without lazy

### PR DESCRIPTION
**Pull request by Mtaylorr**
Issue:
======
When you have duplicate groupbys with `lazy=False` you will get an error.

Steps to reproduce the error:
=============================
- Install timesheet
- Go to timesheet / reporting / By Employee
- Add a groupby by month for one employee
- I will show an error.

Origin of the issue:
====================
The timesheet component will the send a request for read_group having `['date:month', 'date:month']` so we will have duplicated groupby which will be processed later in `_read_group_format_result` which update the value of `row[group]` for each group so the first group will have the original values which is ok but the second one will have the updated values by the first iteration which will result in error.

Solution:
=========
I saved the original values for each group in the row so we can use it later and not have a problem.

opw-3497803

### CODE REVIEW

**Review by nle-odoo — APPROVED** [review]
test looks good

as far as I can tell, the difference between 16.0 and saas-16.3 is that there is not the OrderedSet on the groupby list (that was originally added by be8bbc2396a494a8f22a9d1671891536a7f92abd)

maybe it could be reintroduced (in the read_group) instead of the current version that complicates a little the code

File: odoo/addons/test_read_group/tests/test_duplicate_groupbys.py [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+
+class TestDuplicateGroupBys(common.TransactionCase):
+
+    def test_duplicate_month(self):
+        Model = self.env['res.partner']
+        records = Model.create([
+            {'date': '2022-01-29', 'name': 'Date'}])
+        Model.read_group(
+            [('id', 'in', records.ids)], fields=['date', 'name'], groupby=['date:month', 'date:month'], lazy=False)
+        self.assertTrue(1)
```
Comment (ryv-odoo) [inline odoo/addons/test_read_group/tests/test_duplicate_groupbys.py:14]: `self.assertTrue(1)` is useless. It is better to check the result.

**Review by ryv-odoo — COMMENTED** [review]
Nice catch, thank you @Mtaylorr .
Your fix doesn't to work correctly, I proposed an alternative.
Cheers

File: odoo/models.py [hunk 2]
```diff
@@ -2407,7 +2407,7 @@ def _read_group_format_result(self, rows_dict, lazy_groupby):
 
                 additional_domain = [(field_name, '=', value)]
 
-                if field.type in ('date', 'datetime'):
+                if field.type in ('date', 'datetime') and (isinstance(value, datetime.date) or isinstance(value, datetime.datetime)):
```
Comment (ryv-odoo) [inline odoo/models.py:2410]: `isinstance` can receive tuple of types: => `isinstance(value, (datetime.date, datetime.datetime))`

Sadly, I made a mistake in my proposal, it doesn't take in account `False` value. You can move the extra condition to the next `if` to working as expected.

**Comment by ryv-odoo** [comment]
Thank you @Mtaylorr 
@robodoo r+

**Pull request closed without merging**